pkg/controller/common/helpers.go

```diff
@@ -125,6 +125,19 @@ func MergeMachineConfigs(configs []*mcfgv1.MachineConfig, osImageURL string) (*m
 		}
 	}
 
+	// For layering, we want to let the user override OSImageURL again
+	overriddenOSImageURL := ""
+	for _, cfg := range configs {
+		if cfg.Spec.OSImageURL != "" {
+			overriddenOSImageURL = cfg.Spec.OSImageURL
+		}
+	}
+	// Make sure it's obvious in the logs that it was overridden
+	if overriddenOSImageURL != "" && overriddenOSImageURL != osImageURL {
+		glog.Infof("OSImageURL has been overridden via machineconfig with %s (was: %s)", overriddenOSImageURL, osImageURL)
```

Comment:
maybe we should we also send an event here

Reply:
I moved the logging out a level so the name of the config was available to be logged,  and also added an event.